src/core/csv/reader_parsers.cc

```diff
@@ -573,21 +605,24 @@ void parse_string(FreadTokenizer& ctx) {
 
   // need to skip_whitespace first for the reason that a quoted field might have space before the
   // quote; e.g. test 1609. We need to skip the space(s) to then switch on quote or not.
-  if (*ch==' ' && ctx.strip_whitespace) while(*++ch==' ');  // if sep==' ' the space would have been skipped already and we wouldn't be on space now.
+  if (ch < ctx.eof && *ch==' ' && ctx.strip_whitespace) {
+    while(*++ch==' ' && ch < ctx.eof);  // if sep==' ' the space would have been skipped already and we wouldn't be on space now.
```

Comment:
this doesn't look right: `*++ch==' '` is equivalent to `{ch++; *ch==' '}`. If you compare with eof afterwards, then you can still have an invalid past-the-end-of-file read. Try rewriting it as
```suggestion
if (ctx.strip_whitespace) { 
  while (ch < ctx.eof && *ch == ' ') ch++;
}
```

Reply:
Yeah, you're right. This was an old change and I didn't think it through.